src/Components/Articles.jsx

```diff
@@ -39,34 +49,63 @@ export default function Articles() {
       });
   }, []);
 
+  function handleFilter() {
+    if (filter===false) setFilter(true)
+    else setFilter(false)
+  }
+
   function handleSelectTopic(event) {
-    console.log(event.target.value, "llll")
-    setSelectedTopicRoute(event.target.value);
+    const topic = event.target.value
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("topic", topic);
+    setSearchParams(newParams)
   }
 
-  useEffect(() => {
-    navigate(selectedTopicRoute);
-  }, [selectedTopicRoute]);
+  function handleSelectSortBy(event) {
+    const sortBy = event.target.value
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("sort_by", sortBy);
+    setSearchParams(newParams) 
+  }
 
+  function handleSelectOrder(event) {
+    const order = event.target.value
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set("order", order);
+    setSearchParams(newParams);
+  }
+  
   return (
     <div>
       <ShareButtons />
       {isLoading ? (
         <p>...your articles are loading</p>
       ) : (
         <>
-         
-          <label htmlFor="topics">Pick a topic</label>
-          <select
-            onChange={handleSelectTopic}
-            id="topics"
-          >
-            <option value="/">All topics</option>
+          <label htmlFor="topics">Pick a topic: </label>
+          <select onChange={handleSelectTopic} id="topics" >
+            <option value="">All topics</option>
             {topics.map((topic) => {
               return <TopicOptions key={topic.slug} topic={topic} />;
             })}
           </select>
+             <button onClick={handleFilter}>{filter ? "Close Filter" : "Open filter"}</button>
+        {filter ?
+            <>
+          <label htmlFor="sort-by">Sort by: {}</label>
+          <select onChange={handleSelectSortBy} id="sort-by" >
+            <option value="created_at">date</option>
+            <option value="comment_count">Comment count</option>
+            <option value="votes">Votes</option>
+          </select>
 
+          <label htmlFor="order">order: {}</label>
+          <select onChange={handleSelectOrder} id="order" >
+            <option value="desc">Descending</option>
+            <option value="asc">Ascending</option>
+          </select>
```

Comment:
I reckon all the filtering code could be a sub-component to debloat this component code.

Reply:
done
 thanks looks better now

Reply:
sorted date in both url and options